packages/openspp/test/Adaptor.test.js

```diff
@@ -0,0 +1,100 @@
+import { expect } from "chai";
+import {
+  execute,
+  getServicePoint,
+  searchGroup,
+  searchIndividual,
+  getGroup,
+  getIndividual,
+  getProgram 
+} from "../src/Adaptor.js";
+
+const state = {
+  configuration: {
+    username: "admin",
```

Comment:
Please don't store credentials in source control! For many reasons this is a big no-no. I can happily explain in more detail.

We also don't like to use unit tests which run against live systems. They run slowly, tend to be flaky, and rely on credentials not expiring.

What I was asking for wasn't a unit test, but a test job that would run against OpenSPP's own sandbox. I could run this locally using the sandbox credentials through https://demo.openspp.org (I think?).

Perhaps what  I can do is run these  tests locally as a one-off, just to see that it's all working, and I can accept the PR on that basis.

I'll do that ASAP but I have some very high priority work on my desk this week so I'll have to steal half an hour for it :)

Reply:
oh, this is our test server 😄 the name can be misunderstanding, but I have the permission to test on it

Reply:
_You_ have permission, sure, but you're about to check this code into our public respository. Which means _everyone_ will be testing against it. Every time someone runs the test suite in this repo, it'll call our to your test server with your credentials.

It is critical that these credentials are removed from this PR. I will probably squash the merge into `main` so that it's removed from git history too.